wandb/sdk/artifacts/artifact.py

```diff
@@ -491,9 +490,7 @@ def ttl(self) -> Union[timedelta, None]:
         Raises:
             ArtifactNotLoggedError: Unable to fetch inherited TTL if the artifact has not been logged or saved
         """
-        if self._ttl_is_inherited and (
-            self._state == ArtifactState.PENDING or self._ttl_changed
-        ):
```

Comment:
Wait, why did you change it? This was correct.

Reply:
This caused the following issue:
 ```
a = wandb.Artifact(...)
......
print(a.ttl) # this returns an error even though the user never set a TTL
```
I changed it so it just returns `None` but if its intended behavior to error even when TTL was never set I can change it back

Reply:
Yes, I think the error is intended. If you don't set the ttl, it will be inherited from the collection. So until you save the artifact, we don't know what the ttl is going to be.